src/modules/plus/filter_dynamicgpstext.c

```diff
@@ -0,0 +1,1714 @@
+			crt_point->calc_d_elev = total_d_elev;
+			crt_point->calc_elev_up = total_elev_up;
+			crt_point->calc_elev_down = total_elev_down;
+			crt_point->calc_dist_up = total_dist_up;
+			crt_point->calc_dist_down = total_dist_down;
+			crt_point->calc_dist_flat = total_dist_flat;
+		}
+
+		prev_point = crt_point;
+	}
+	//_DEBUG_write_smoothed_gps_to_file(filter, req_smooth);
+}
+
+
+/** Process a .tcx gps location file, result is put in gps_points[],
+ * ignores points without strictly increasing time (or 0)
+*/
+static void process_tcx(mlt_filter filter, char* file_content, int bytes_read)
+{
+	private_data* pdata = (private_data*)filter->child;
+
+	// sample point from .TCX file, only lat and lon are mandatory but we also need time for syncing
+	/*
+	  <Trackpoint>
+		<Time>2021-02-27T14:03:02+00:00</Time>
+		<Position>
+		  <LatitudeDegrees>44.48205950925148</LatitudeDegrees>
+		  <LongitudeDegrees>26.1843781367422</LongitudeDegrees>
+		</Position>
+		<AltitudeMeters>65.0</AltitudeMeters>
+		<DistanceMeters>25919.213486999884</DistanceMeters>
+		<HeartRateBpm xsi:type="HeartRateInBeatsPerMinute_t">
+		  <Value>124</Value>
+		</HeartRateBpm>
+	  </Trackpoint>
+	*/
+	gps_point crt_point;
+	char* ptext = file_content;
+	const char* last_address = file_content+bytes_read;
+	int i;
+	
+//	mlt_log_info(NULL, "filter_dynamicgps.c entered process_tcx, ptext=%p, last_address=%p", ptext, last_address);
+//	mlt_log_info(NULL, "filter_dynamicgps.c entered process_tcx, ptext=%.100s", ptext);
+	
+	//ptext will point to the current location in the string where we're processing stuff
+	//we will first look for the start of an entry (<Trackpoint>) then try to parse all tags <> </> until the end tag </Trackpoint>
+		
+	//before actual processing takes place, we need to know how many points there are in the file so we can allocate exact gps_points
+	int nr_points = 0, crt_nr_point = 0, skipped_points = 0;
+	while (ptext < last_address)
+	{
+		if (strncmp(ptext, "<Trackpoint>", 12)==0)
+			nr_points++;
+		ptext++;
+	}
+	//mlt_log_info(NULL, "filter_dynamicgps.c found %d gps points in file", nr_points);
+	
+	if (pdata->gps_points)
+		free(pdata->gps_points);
+	pdata->gps_points = malloc(sizeof(gps_point) * nr_points);
+	if (!pdata->gps_points)
+		return;
+	pdata->gps_points_size = nr_points;
+	
+	//move pointer back to beginning
+	ptext = file_content;
+	
+	//find the start of <Trackpoint> field and jump to it
+	while (ptext<last_address && (ptext = strstr(ptext, "<Trackpoint>")) != NULL)
+	{
+		crt_point = uninit_gps_point;
+		
+		//iterate through the entire Trackpoint tag and find sub-tags
+		for (ptext += strlen("<Trackpoint>"); strncmp(ptext, "</Trackpoint>", 13) && ptext < last_address; ptext++)
+		{
+			if (strncmp(ptext, "<Time>", 6) == 0)
+			{
+				//skip tag
+				ptext+=6;
+				//format: YYYY-MM-DDThh:mm:ss (+timezone optional?)
+				if (ptext+19 >= last_address) {
+					mlt_log_warning(NULL, "filter_dynamicgps.c <time> tag can't fit time, reached EOF");
```

Comment:
Pass the filter into the log function and it will automatically add the file name.

Reply:
I did not know that, will update.